jit/jit-block.c

```diff
@@ -855,6 +916,32 @@ _jit_block_clean_cfg(jit_function_t func)
 				block->ends_in_dead = 1;
 				delete_edge(func, block->succs[1]);
 			}
+			else if(block->num_succs == 2
+				&& is_empty_block(block->next)
+				&& block->next->num_succs == 1
+				&& block->next->succs[0]->flags == _JIT_EDGE_BRANCH
+				&& block->succs[0]->dst == block->next->next)
+			{
+				/* We have a conditional branch, that
+				   branches around the next block, and
+				   the next block consists of just a
+				   jump, like:
+
+					if l7 != 3 then goto .L0
+					goto .L1
+					.L0:
+
+				   In this case we can invert the
+				   condition and retarget the jump.  */
+				insn->opcode = _jit_invert_condition(insn->opcode);
+				detach_edge_dst(block->succs[0]);
+				attach_edge_dst(block->succs[0], block->next->succs[0]->dst);
+				insn->dest = (jit_value_t) block->next->succs[0]->dst->label;
+				detach_edge_dst(block->next->succs[0]);
+				attach_edge_dst(block->next->succs[0], block->next->next);
+				block->next->succs[0]->flags = _JIT_EDGE_FALLTHRU;
+				changed = 1;
```

Comment:
If I understand correctly what you do here then to maintain consistency additional changes are needed: the unconditional branch opcode should be changed to the noop opcode and the block->next->ends_in_dead flag should be reset. But this is irrelevant in the light of the previous comment.